src/distilabel/embeddings/llamacpp.py

```diff
@@ -0,0 +1,179 @@
+                " `pip install llama-cpp-python`."
+            ) from ie
+
+        if self.hub_repository_id is not None:
+            try:
+                from huggingface_hub.utils import validate_repo_id
+
+                validate_repo_id(self.hub_repository_id)
+            except ImportError as ie:
+                raise ImportError(
+                    "Llama.from_pretrained requires the huggingface-hub package. "
+                    "You can install it with `pip install huggingface-hub`."
+                ) from ie
+            try:
+                self._logger.info(
+                    f"Attempting to load model from Hugging Face Hub: {self.hub_repository_id}"
+                )
+                self._model = _LlamaCpp.from_pretrained(
+                    repo_id=self.hub_repository_id,
+                    filename=self.model,
+                    verbose=self.verbose,
+                    embedding=True,
+                    kwargs=self.extra_kwargs,
+                )
+                self._logger.info("Model loaded successfully from Hugging Face Hub")
+            except Exception as e:
+                self._logger.error(
+                    f"Failed to load model from Hugging Face Hub: {str(e)}"
+                )
+                raise
+        else:
+            try:
+                self._logger.info(f"Attempting to load model from: {self.model_name}")
+                self._model = _LlamaCpp(
+                    model_path=self.model_name,
+                    verbose=self.verbose,
+                    embedding=True,
+                    kwargs=self.extra_kwargs,
+                )
+                self._logger.info(f"self._model: {self._model}")
+                self._logger.info("Model loaded successfully")
+            except Exception as e:
+                self._logger.error(f"Failed to load model: {str(e)}")
+                raise
+
+    def unload(self) -> None:
+        """Unloads the `gguf` model."""
+        CudaDevicePlacementMixin.unload(self)
+        super().unload()
+
+    @property
+    def model_name(self) -> str:
+        """Returns the name of the model."""
+        return self.model
+
+    def encode(self, inputs: List[str]) -> List[List[Union[int, float]]]:
+        """Generates embeddings for the provided inputs.
+
+        Args:
+            inputs: a list of texts for which an embedding has to be generated.
+
+        Returns:
+            The generated embeddings.
+        """
+        if self._model is None:
+            self._logger.error("Model is not initialized")
+            raise ValueError(
+                "Model is not initialized. Please check the initialization process."
+            )
+
+        try:
+            embeds = self._model.embed(inputs, normalize=self.normalize_embeddings)
+            return embeds
+        except Exception as e:
+            print(f"Error creating embedding: {str(e)}")
```

Comment:
the try except is not needed and always avoid print statements

Reply:
@davidberenstein1957 : incorporated the feedback. In addition will keep the a note `avoid print statements` to all my further PRs.